### ISSUE TITLE
Updated analysis: update `fusiong_filtering` module to handle gene symbols that have no mapped ENSG ID

### ISSUE THREAD

**Issue by logstar** [PediatricOpenTargets/ticket-tracker#153]
#### What analysis module should be updated and why?

The `fusiong_filtering` module should be updated to handle gene symbols that have no mapped ENSG ID.

There are some gene symbols that have no mapped ENSG ID in `analyses/fusion_filtering/results/fusion-putative-oncogenic.tsv`, as @kgaonkar6 pointed out at <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/49#issuecomment-889450851>. The following gene symbols are currently the only ones that have no mapped ENSG ID: IGH-@, IGH@, IGL-@, and IGL@. (The code for checking them is listed below.) These gene symbols are [IGL immunoglobulin lambda locus](https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=3535) and [IGH immunoglobulin heavy locus](https://www.ncbi.nlm.nih.gov/gene/3492), and there is no Ensembl link out from the NCBI gene pages. These gene symbols are introduced by starfusion, <https://github.com/STAR-Fusion/STAR-Fusion/issues/197#issuecomment-597215303>, but it is unclear whether there is any difference between IGH-@ and IGH@, or between IGL-@ and IGL@.

```text
> f_tbl <- read_tsv('../../fusion-frequencies/results/putative-oncogene-fusion-freq.tsv.gz')
> f_gene_tbl <- read_tsv('../../fusion-frequencies/results/putative-oncogene-fused-gene-freq.tsv.gz')
> map_tbl <- read_tsv('../../../data/ensg-hugo-rmtl-mapping.tsv')
> f_tbl %>% filter(!Gene_Symbol %in% map_tbl$gene_symbol) %>% group_by(Gene_Symbol) %>% tally()
# A tibble: 4 x 2
  Gene_Symbol     n
  <chr>       <int>
1 IGH-@           3
2 IGH@           10
3 IGL-@           2
4 IGL@            2
> f_gene_tbl %>% filter(!Gene_Symbol %in% map_tbl$gene_symbol) %>% group_by(Gene_Symbol) %>% tally()
# A tibble: 4 x 2
  Gene_Symbol     n
  <chr>       <int>
1 IGH-@           9
2 IGH@           44
3 IGL-@           4
4 IGL@            4
```

These gene symbols are also confirmed to have no mapped ENSG IDs on NCBI gene, mygene.info, HGNC, and biomart:

- IGH
  - NCBI gene: <https://www.ncbi.nlm.nih.gov/gene/3492>
  - mygene.info: <http://mygene.info/v3/gene/3492>
  - HGNC: <https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:5477>
  - biomart: see code below
- IGL
  - NCBI gene: <https://www.ncbi.nlm.nih.gov/gene?Db=gene&Cmd=DetailsSearch&Term=3535>
  - mygene.info: <http://mygene.info/v3/gene/3535>
  - HGNC: <https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:5853>
  - biomart: see code below

```text
> bm_ensembl <- biomaRt::useEnsembl(
+   biomart = 'ensembl', dataset = 'hsapiens_gene_ensembl')
> biomaRt::getBM(
+   attributes = c('ensembl_gene_id', 'description'),
+   filters = c('hgnc_id'),
+   values = c(
+     'HGNC:5853', # IGL
+     'HGNC:5477', # IGH
+     'HGNC:5876' # this should have ENSG ID, to show this code works
+     ),
+   mart = bm_ensembl)
  ensembl_gene_id                                                            description
1 ENSG00000273737 immunoglobulin lambda variable 1-36 [Source:HGNC Symbol;Acc:HGNC:5876]
2 ENSG00000211655 immunoglobulin lambda variable 1-36 [Source:HGNC Symbol;Acc:HGNC:5876]
```

However, the IGH and IGL loci have gene groups on HGNC.

- IGH gene group: <https://www.genenames.org/data/genegroup/#!/group/349>
- IGL gene group: <https://www.genenames.org/data/genegroup/#!/group/353>

The IGH and IGL loci are relevant to lymphoma, according to <https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/igl> and <https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/igh>, so we need to find their corresponding gene ENSG IDs, in order to make them available on the PedOT website, as PedOT uses ENSG IDs for navigation, according to slack message <https://opentargetspediatrics.slack.com/archives/C021Z53SK98/p1627847958027600>.

#### What changes need to be made? Please provide enough detail for another participant to make the update.

According to @kgaonkar6's comment at <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/49#discussion_r680161050>:

> We've noticed this issue with IGH@ symbols before and had to manually retain/annotate these fusions with IGH*@ as oncogenic fusions in OpenPBTA and current `fusion_filtering` modules.
>
> When we checked (last year I think) there were multiple IGH genes around the breakpoint and gtf, we don't have the IGH*@ genes in the gtf file we use to get the ENSG ID so you are right will have to discuss and manually add the matched ENSG ID to `ensg-hugo-rmtl-mapping.tsv`.

Following are potential approaches to handle the gene symbols that have no mapped ENSG ID, but this issue is open for discussions and suggestions on other approaches.

- If starfusion also outputs actual IGH/IGL genes fusions, e.g. [IGHE ENSG00000211891](https://useast.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000211891;r=14:105597691-105601728) as additional rows, maybe we could remove these symbols. Also add note and code to check these are the only gene symbols without ENSG ID.
- If starfusion does not output corresponding IGH/IGL genes, maybe we could infer their gene ENSG IDs from the break points.
- If these IGH/L@ fusions really have no corresponding ENSG IDs and are useful, maybe we could discuss with the FNL team on handling these gene symbols that have no ENSG ID on the PedOT website in the `#ot-portal-content` channel.

#### What input data should be used? Which data were used in the version being updated?

`fusiong_filtering` module input data.

#### When do you expect the revised analysis will be completed?

2-5 days

#### Note

This issue is a summary of the discussions in <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/49>.

cc @kgaonkar6 @jharenza

**Comment by jharenza** [comment]
@sangeetashukla - since this has just come up in #264, we should revisit this ticket. Can you work on the first two bullets outlined here? I don't think it will be super urgent, so you can prioritize based on your other ongoing tasks - @zdorman can confirm. @runjin326 can help advise on this if needed. Thanks! Cc @afarrel

**Comment by zdorman** [comment]
Apologies for reporting #264 when you have already clearly outlined the issue here! 
* Agreed that this is not a major/urgent issue and can wait until the next release
* I checked all of the v10 JSONL files and found no other instances of blank ENSGs
* MTP (formerly PedOT) is extremely ENSG-centric, so without an ENSG the record is functionally nonexistent
* Agreed that the best solution is to remove the records if the data is captured elsewhere in starfusion outputs
    * If not, then we could consider a "best match" approach to manually assign ENSGs
    * If no best match is appropriate, we can remove the records. The fusion frequencies will still be accessible to a user from the perspective of the other fused gene(s)

@jharenza @sangeetashukla @runjin326 @afarrel

**Comment by sangeetashukla** [comment]
> @sangeetashukla - since this has just come up in #264, we should revisit this ticket. Can you work on the first two bullets outlined here? I don't think it will be super urgent, so you can prioritize based on your other ongoing tasks - @zdorman can confirm. @runjin326 can help advise on this if needed. Thanks! Cc @afarrel

@jharenza Sure, I will look into this issue.

**Comment by jharenza** [comment]
closed with #162
